api/maven-api-core/src/main/java/org/apache/maven/api/Dependency.java

```diff
@@ -23,30 +23,45 @@
 import org.apache.maven.api.annotations.Nonnull;
 
 /**
+ * Result of resolving a {@code DependencyCoordinate}.
+ * Resolving is the process that clarifies the obligation (optional or mandatory status),
+ * selects a particular version and downloads the artifact in the local repository.
```

Comment:
It's a bit more complicated as explained earlier.  `Dependency` is the output of the `collection` process, which builds the graph of dependencies. It then needs to be `flattened` and `resolved`.  The version selection is done for each dependency during the collection phase, the flatten phase will keep only a single version per GA (groupId+artifactId), the resolution will actually download the dependencies (or artifacts) that have been computed.